modules/wallet/update.go

```diff
@@ -300,6 +297,12 @@ func (w *Wallet) ProcessConsensusChange(cc modules.ConsensusChange) {
 		w.log.Println("ERROR: failed to update consensus change ID:", err)
 	}
 
+	if height, err := dbGetConsensusHeight(w.dbTx); err == nil && height%100 == 0 {
+		// sync every 100 blocks to prevent running transaction from growing
+		// too large
+		w.syncDB()
+	}
```

Comment:
I'm pretty sure you already fetch the height at some point in one of the above helper functions. Maybe you can pass it through to this point?

Also, there's a big miss here. You aren't counting blocks, you are checking the height. A consensus change can include multiple blocks, and I know at least one contributor is working on code that intends to lump multiple blocks together (resulting in a 60% speedup for IBD). If those blocks come 5 at a time with the offset 51, 56, 61, etc., then this will never trigger.